packages/workflow/src/interceptors.ts

```diff
@@ -20,68 +18,212 @@ import { ChildWorkflowOptionsWithDefaults, ContinueAsNewOptions } from './interf
 
 export { Next, Headers };
 
-/** Input for WorkflowInboundCallsInterceptor.execute */
+/**
+ * A function that instantiates {@link WorkflowInterceptors}.
+ *
+ * Workflow interceptor modules should export an `interceptors` function of this type.
+ *
+ * @example
+ *
+ * ```ts
+ * export function interceptors(): WorkflowInterceptors {
+ *   return {
+ *     inbound: [],   // Populate with list of inbound interceptor implementations
+ *     outbound: [],  // Populate with list of outbound interceptor implementations
+ *     internals: [], // Populate with list of internals interceptor implementations
+ *   };
+ * }
+ * ```
+ */
+export type WorkflowInterceptorsFactory = () => WorkflowInterceptors;
+
+/**
+ * A mapping from interceptor type to an optional list of interceptor implementations
+ */
+export interface WorkflowInterceptors {
+  inbound?: WorkflowInboundCallsInterceptor[];
+  outbound?: WorkflowOutboundCallsInterceptor[];
+  internals?: WorkflowInternalsInterceptor[];
+}
+
+// Workflow Inbound Calls Interceptors /////////////////////////////////////////////////////////////////////////////////
```

Comment:
nit: comment format is not consistent with the Activity/Worker interceptor formats